internal/plugins/helm/init.go

```diff
@@ -0,0 +1,212 @@
+	"github.com/operator-framework/operator-sdk/internal/scaffold/input"
+	"github.com/operator-framework/operator-sdk/internal/util/projutil"
+)
+
+type initPlugin struct {
+	// For help text.
+	commandName string
+
+	// Helm flags
+	APIVersion       string
+	Kind             string
+	CRDVersion       string
+	HelmChartRef     string
+	HelmChartVersion string
+	HelmChartRepo    string
+	gitInit          bool
+	SkipGeneration   bool
+}
+
+var (
+	_ plugin.Init        = &initPlugin{}
+	_ cmdutil.RunOptions = &initPlugin{}
+)
+
+func (p *initPlugin) UpdateContext(ctx *plugin.Context) {
+	ctx.Description = `Initialize a Helm project with the Helm Chart package directories.`
+	ctx.Examples = fmt.Sprintf(`  # Scaffold a Helm project %s 
+	
+  $ operator-sdk init --plugins=helm.kubebuilder.io/v1-alpha \
+  --api-version=app.example.com/v1alpha1 \
+  --kind=AppService
+
+  $ operator-sdk init --plugins=helm.kubebuilder.io/v1-alpha \
+  --api-version=app.example.com/v1alpha1 \
+  --kind=AppService \
+  --helm-chart=myrepo/app
+
+  $ operator-sdk init --plugins=helm.kubebuilder.io/v1-alpha \
+  --helm-chart=myrepo/app
+
+  $ operator-sdk init --plugins=helm.kubebuilder.io/v1-alpha \
+  --helm-chart=myrepo/app \
+  --helm-chart-version=1.2.3
+
+  $ operator-sdk init --plugins=helm.kubebuilder.io/v1-alpha \
+  --helm-chart=app \
+  --helm-chart-repo=https://charts.mycompany.com/
+
+  $ operator-sdk init --plugins=helm.kubebuilder.io/v1-alpha \
+  --helm-chart=app \
+  --helm-chart-repo=https://charts.mycompany.com/ \
+  --helm-chart-version=1.2.3
+
+  $ operator-sdk init --plugins=helm.kubebuilder.io/v1-alpha \
+  --helm-chart=/path/to/local/chart-directories/app/
+
+  $ operator-sdk init --plugins=helm.kubebuilder.io/v1-alpha \
+  --helm-chart=/path/to/local/chart-archives/app-1.2.3.tgz
+	"
+`,
+		ctx.CommandName)
+
+	p.commandName = ctx.CommandName
+}
+
+func (p *initPlugin) BindFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&p.APIVersion, "api-version", "",
+		"Kubernetes apiVersion and has a format of $GROUP_NAME/$VERSION (e.g app.example.com/v1alpha1)")
+	fs.StringVar(&p.Kind, "kind", "",
+		"Kubernetes resource Kind name. (e.g AppService)")
+	fs.BoolVar(&p.SkipGeneration, "skip-generation", false,
+		"Skip generation of deepcopy and OpenAPI code and OpenAPI CRD specs")
+	fs.StringVar(&p.CRDVersion, "crd-version", gencrd.DefaultCRDVersion,
+		"CRD version to generate")
+	fs.StringVar(&p.HelmChartRef, "helm-chart", "",
+		"Initialize helm operator with existing helm chart (<URL>, <repo>/<name>, or local path).")
+	fs.StringVar(&p.HelmChartVersion, "helm-chart-version", "",
+		"Specific version of the helm chart (default is latest version)")
+	fs.StringVar(&p.HelmChartRepo, "helm-chart-repo", "",
+		"Chart repository URL for the requested helm chart")
+	fs.BoolVar(&p.gitInit, "git-init", false,
+		"Initialize the project directory as a git repository (default false)")
+}
+
+func (p *initPlugin) InjectConfig(c *config.Config) {
+	//todo: implement config
```

Comment:
I don't understand why
```suggestion
	c.Layout = plugin.KeyFor(Plugin{})
	c.config = c
```
wouldn't work now. Can you elaborate on what you meant in [your comment](https://github.com/operator-framework/operator-sdk/pull/3193/files#r438700264) below?

Reply:
The above comment has no relation with it. We can indeed add here the config implementation. However, was decided that it should not be made now so it was not added. 
IMO we could add all config now as well. 

However, @joelanford said that it would raise discussions over the name/version of the plugin and it is out of the task scope. So, the other comment is because the suggested we indeed now cause panic when the getName and get Version is called. However, it is not possible.

Reply:
If having a plugin name is outside of the scope of this PR, can you remove the plugin from `cli.New` so it isn't exposed?

Reply:
If we remove we cannot test it alone/manually.

Reply:
If we remove from cli we cannot run the plugin alone to check it which IMO is very good for we move forward with it and b able to test the future changes. So, a see few options.

1) Remove the plugin from the cli and apply the follow-up PR to bind the flags here in order to allow the CI to check it. 
2) Add the config on it since it is really easy to get done and requires just a few lines. 
3) Keep it as it is now which allow test manually and do not add the config. 

@joelanford @estroz how would like to move forward here?

Reply:
I would prefer 2. Also I think the name `helm.operator-sdk.io` is pretty straightforward, not sure why we _wouldn't_ want to use this name.

Reply:
Me too. Are you ok with the version v1-alpha?

Reply:
Done: 

```
 $ cat PROJECT 
layout: helm.kubebuilder.io/v1-alpha
version: 3-alpha
plugins:
  helm.kubebuilder.io/v1-alpha: {}

```